packages/@aws-cdk/aws-eks/lib/cluster-resource-handler/common.ts

```diff
@@ -39,15 +39,13 @@ export abstract class ResourceHandler {
       RoleSessionName: `AWSCDK.EKSCluster.${this.requestType}.${this.requestId}`,
     });
 
-    const proxyAddress = this.httpProxyFromEnvironment();
-    if (proxyAddress) {
-      this.log(`Using proxy server: ${proxyAddress}`);
-      // eslint-disable-next-line @typescript-eslint/no-require-imports, import/no-extraneous-dependencies
-      const ProxyAgent: any = require('proxy-agent');
-      aws.config.update({
-        httpOptions: { agent: new ProxyAgent(proxyAddress) },
-      });
-    }
+    // Configure the proxy agent. By default, this will use HTTPS?_PROXY and
+    // NO_PROXY environment variables to determine which proxy to use for each
+    // request.
+    //
+    // eslint-disable-next-line @typescript-eslint/no-require-imports, import/no-extraneous-dependencies
+    const ProxyAgent: any = require('proxy-agent');
```

Comment:
I believe this change is causing EKS deployments to fail. CloudFormation now throws `Internal Failure.` when trying to create. 

This handler is used in the "onComplete" Lambda function which does not have the `proxy-agent` available. We only pass an `http_proxy` env var into the "onEvent" Lambda since that is the only Lambda that makes requests with the `aws-sdk-js`.

I tested this by moving the proxy logic to the `onEvent()` method and was able to successfully deploy again. I'll make a PR.

Reply:
I also noticed that `new ProxyAgent()` does not automatically detect `process.env.http_proxy` or `process.env.HTTP_PROXY`.

Reply:
Here's what ProxyAgent looks like when you log it in the Lambda:

Code:
```ts
const agent = new ProxyAgent()
console.log(`agent: ${JSON.stringify(agent, null, 2)}`);
```
Logs:
```log
2021-10-01T22:32:09.812Z	03de5aeb-e937-4b3d-91a3-aee2c524cb47	INFO	agent: {}
```

---

Code:
```ts
const agent = new ProxyAgent(process.env.http_agent);
console.log(`agent: ${JSON.stringify(agent, null, 2)}`);
```
Logs:
```log
2021-10-01T22:32:09.812Z	03de5aeb-e937-4b3d-91a3-aee2c524cb47	INFO	agent: {}
```

Reply:
Tested using https://github.com/ryparker/aws-cdk-sample-eks

Reply:
It's normal and expected that the `ProxyAgent` instance does not look the same. If provided with a proxy, it will use ALWAYS and ONLY that proxy. When not specified, it will delegate to `proxy-from-env`, which will look at the correct proxy environment variable for the request' protocol & honor NO_PROXY.

I'd argue that the broken Lambda function should receive an `HTTPS_PROXY` and not `HTTP_PROXY`, as the AWS SDK typically makes only TSL connections... So it should be expected NOT to use `HTTP_PROXY` (using that value is incorrect there and should be considered a bug).